packages/frontend/src/hooks/useUsdValue.ts

```diff
@@ -0,0 +1,31 @@
+import Big from 'big.js';
+import { useEffect, useState } from 'react';
+import { useSifi } from 'src/providers/SDKProvider';
+
+type UseUsdPriceOptions = {
+  address?: string | null;
+  chainId?: number | null;
+  amount?: string;
+};
+
+const useUsdValue = ({ address, chainId, amount }: UseUsdPriceOptions) => {
+  const [usdPrice, setUsdPrice] = useState<string>();
+  const sifi = useSifi();
+
+  const fetchUsdPrice = async () => {
+    if (!address || !chainId) return;
+
+    const response = await sifi.getUsdPrice(chainId, address);
+    setUsdPrice(response.usdPrice);
+  };
+
+  useEffect(() => {
+    fetchUsdPrice();
+  }, [address, chainId]);
+
+  if (!usdPrice || !amount) return '';
+
+  return Big(Number(usdPrice) * Number(amount)).toFixed(2);
```

Comment:
Not sure I understand why we need to convert with Big.js here.
If we wanted to use bignumbers for safe multiplication, we'd want to be doing the conversion before - but given the output is a USD trade value, and we are not dealing with wei/satoshi here, we shouldn't be running into accuracy issues.

Reply:
Oh I see, amount could be high precision right - if a user has clicked "max" for example?

In that case the issue would still lie in the inner multiplication.

Reply:
Big.js toFixed function is better than the native Number's toFixed as it does not add unnecessary decimals.